### PR TITLE
fix: pend bounds change when moving BrowserWindows

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/33279.
Refs CRBUG:251813

It's the case at present that `widget()->SetBounds(bounds)` does not take effect if the window is currently being moved or resized. This is an open issue on Chromium since 2013, with no real solution implemented upstream but several options suggested, including:

> Drag completes ignoring the bounds update but the bounds update is queued and then executed after drag completes.

In a Chromium context, this might not be ideal, but in an Electron context I believe this is the most expected behavior from a development standpoint. As a result of this PR, the bounds update correctly once the user completes the resize or the drag movements to reflect the new bounds set programmatically. Now, we take the last call to `setBounds` and apply it once the user has finished the move or drag operation.

Tested with https://gist.github.com/6b6ea5cdaa1883656a8bd0bf89195283.

This is unfortunately not directly testable via spec, given that it's dependent on user-initiated movement.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where new bounds set via `setBounds` was not correctly applied if the user was moving or resizing the window concurrently on Windows.

### CODE REVIEW

**Comment by dsanders11** [comment]
A couple of questions and comments. First, it looks like this is only implemented for Windows? If that's intentional, can it be mentioned in the description and release notes?

- If Windows only, can we `#ifdef` the changes? It looks to me like with the current code the queue will be filled on Linux and `SetBounds` while moving/resizing ignored and the queue will never be read
- This is implemented as a queue, but only one item is popped when the move/resize ends. What happens if you change the repro gist to use `setinterval` instead of `settimeout`, and keep dragging during that time? Won't multiple bounds changes be queued up, but only one read? Seems that would affect future moves/resizes - it'll start playing back the full queue, but only each time you move/resize the window.

**Comment by codebytere** [comment]
@dsanders11 you're right, i ifdef'd the queue itself and whiffed on the queue fill part. I've also corrected the queue pop logic.

**Comment by pushkin-** [comment]
Is there a reason we even need a queue, or should we just remember whatever the latest setBounds call was and just apply that? Why do we need to move the window 20 or whatever times in a row?

**Comment by codebytere** [comment]
@pushkin- i also thought about that. i'd be fine with either.

File: shell/browser/native_window_views.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "shell/browser/native_window.h"
 
 #include <memory>
+#include <queue>
```
Comment (jkleinsc) [inline shell/browser/native_window_views.h:11]: It doesn't look like this is used?

**Review by jkleinsc — CHANGES_REQUESTED** [review]
It looks like true queuing wasn't implemented.  Not sure if it was intentional or not.

**Review by jkleinsc — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**